crates/rpc/rpc-eth-api/src/core.rs

```diff
@@ -618,11 +619,11 @@ where
     /// Handler for: `eth_simulateV1`
     async fn simulate_v1(
         &self,
-        opts: SimBlock,
+        payload: SimulatePayload,
         block_number: Option<BlockId>,
-    ) -> RpcResult<Vec<SimulatedBlock>> {
+    ) -> RpcResult<Vec<SimulatedBlock<Block<WithOtherFields<reth_rpc_types::Transaction>>>>> {
```

Comment:
why does this not return `B`, because this response should include txs with additional fields?

Reply:
our primivie -> rpc conversions return such `Block` so I've just reused it https://github.com/paradigmxyz/reth/blob/875cee27aca747315af1ef2a9dcdf29e058b33e7/crates/rpc/rpc-types-compat/src/block.rs#L18-L23

I think this generic type should be same to the network's block response in general